### ISSUE TITLE
Test Successful Creation of Both Printer Task Definition and Printer Task Trigger Using Permissions from Single AppId

### ISSUE THREAD

**Issue by fey101** [microsoftgraph/msgraph-sdk-raptor#909]
Related to closed issue #https://github.com/microsoftgraph/microsoft-graph-docs/issues/14774

**Description**
------------------
Get/Create PrintTaskDefinition strictly require application permission whereas the get/create PrintTaskTrigger endpoints strictly require delegated permissions. A creation of a task trigger requires a task definition reference in its body. 

In raptor, to avoid the maximum permission limit, we use a single default app for all application permissions and then create delegated apps for each delegated permission. This creates a situation where **the appId is different** in the tokens used to create task definition and that used to create a taskTrigger resulting to the error "message: Invalid Operation:No Task definition object found with given parameters." even when a `Get` to a task definition returns a list of definitions. The assumption is, the same reason would cause the initial error which was "message": "Invalid Operation: There is a task trigger already created for the printer <printer_id>. Multiple task triggers per printer is not supported." even when an initial GET query had indicated no task Triggers exist.

** Action Items**
- [ ] Create a Single app in a test tenant with all required printer and related resources permissions
- [ ] Test creation of a task trigger whereby the token/tokens used to create the printer, printer task definition, and printer task trigger have reference to the same tenantId and appId 
- [ ] If successful figure out how to replicate this scenario in our prod tenant

**Comment by zengin** [comment]
@fey101 please let me know if I misunderstood, does this mean that a task definition created by one app registration is not accessible by another app registration even if they have the same set of permissions?

**Comment by fey101** [comment]
> @fey101 please let me know if I misunderstood, does this mean that a task definition created by one app registration is not accessible by another app registration even if they have the same set of permissions?

Yes. It appears so. But I can explicitly confirm this from @nilakhan

**Comment by nilakhan** [comment]
Yes, thats correct. Task definitions are created using application permissions and are owned by applications. 
Universal Print provides a platform for 3P apps to insert hooks in the print job pipeline. 
Task definitions are kind of hook definition that is owned by the application that created it.

**Comment by zengin** [comment]
@nilakhan, can this limitation be documented under the Permissions section of the docs: https://docs.microsoft.com/en-us/graph/api/printer-list-tasktriggers?view=graph-rest-1.0&tabs=http#permissions?

Also wondering, is this a deliberate design choice? Because it sounds like a "Printer Administrator" with "Printer.FullControl.All" permissions won't be able to access a trigger that exists in the system, unless they use the same appId that created the trigger.
